src/Featurizers/Components/GrainFeaturizerImpl.h

```diff
@@ -499,19 +501,24 @@ Impl::GrainEstimatorImplBase<BaseT, GrainT, EstimatorT, MaxNumTrainingItemsV>::G
 }
 
 template <typename BaseT, typename GrainT, typename EstimatorT, size_t MaxNumTrainingItemsV>
-Impl::GrainEstimatorImplBase<BaseT, GrainT, EstimatorT, MaxNumTrainingItemsV>::GrainEstimatorImplBase(char const *name, AnnotationMapsPtr pAllColumnAnnotations, CreateEstimatorFunc createFunc) :
-    BaseT(name, pAllColumnAnnotations),
-    _pAllColumnAnnotations(pAllColumnAnnotations),
-    _createFunc(
-        std::move(
-            [&createFunc](void) -> CreateEstimatorFunc const & {
-                if(!createFunc)
-                    throw std::invalid_argument("createFunc");
+Impl::GrainEstimatorImplBase<BaseT, GrainT, EstimatorT, MaxNumTrainingItemsV>::GrainEstimatorImplBase(AnnotationMapsPtr pAllColumnAnnotations, CreateEstimatorFunc createFunc) :
+    BaseT(
+        [pAllColumnAnnotations, &createFunc](void) -> std::string {
+            if(!createFunc)
+                throw std::invalid_argument("createFunc");
 
-                return createFunc;
-            }()
-        )
+            // Create the name of the GrainEstimator as "Grain<EstimatorName>". This is slightly more
+            // complicated than what would normally be expected, as we have to create an instance
+            // of the estimator before we can get its name.
+
+            EstimatorT const                estimator(createFunc(std::move(pAllColumnAnnotations)));
+
+            return std::string("Grain") + estimator.Name;
+        }(),
+        pAllColumnAnnotations

```

Comment:
Do we need to worry about moving `pAllColumnAnnotations` on line 514 and then using it again on line 518?

Reply:
In this case, we don't as the lambda makes a copy of the value in its reference list on line 506.